edgecompute/examples/traffic-routing/url-shortner/main.js

```diff
@@ -0,0 +1,147 @@
+
+  // Request path
+  const path = request.path;
+
+  // If the request path contains /r/ then we are serving a redirect
+  if (path.split('/')[1] === 'r') {
+    let key = path.split('/')[2];
+    
+    // Retrieve a redirect URL associated with the key
+    let redirectUrl;
+    try {
+        redirectUrl = await edgeKv.getText({ item: key, default_value: undefined });
+        logger.log('checking ID ' + key + ".  EdgeKV contained: "+ redirectUrl );
+    } catch (error) {        
+        logger.log("Error serving redirect: " + error);         
+    }    
+
+    // If the redirect exists, return a 302 response.  Otherwise server a 404 error
+    if (redirectUrl != undefined && redirectUrl != null) {
+      return Promise.resolve(createResponse(302, {'Location': [redirectUrl] }, '')); 
+    } else {
+      return Promise.resolve(createResponse(404, {'Content-Type': ['text/html'] }, '<html><body>Redirect not found</body></html>')); 
+    }
+  } 
+
+// If the request path contains /m/ then we are managing redirects
+if (path.split('/')[1] === 'm') {
+  var params = new URLSearchParams(request.query);
+  var name = params.get('name');
+  var url = params.get('url');
+  var status; 
+  var body;  
+  var checkID;
+  
+  // Supported operations are 'add', 'update' and 'delete'
+  switch (params.get('op')) {
+    case 'add':
+      // If a custom redirect name was not defined, then randomly create a name to use as the key
+      if (name === undefined || name === null) {
+        name = Math.random().toString(36).substr(2, 10);
+      }
+
+      // Add redirect to EKV
+      try {
+        checkID = await edgeKv.getText({ item: name, default_value: undefined });
+
+        if(checkID != undefined) {
+          status = "Error - Redirect already exists for /r/" + name;  
+          body = '<b>' + status + '<b><br>';
+          logger.log(status);
+        } else {
+          await edgeKv.putText({ item: name, value: url });
+          status = "Redirect Successfully added";
+          body = '<b>' + status + '<b><br> \
+            <p>URL: <a href="https://' +request.host + '/r/' + name + ' ">https://' +request.host + '/r/' + name +'</a></p>\
+            <p>Redirects to: <a href=" ' + url + ' ">' + url +'</a></p>'; 
+        }
+      } catch (error) {        
+        status = "error creating redirect";        
+        body = '<b>' + status + '<b><br>';
+        logger.log(status);
+      }
+  
+      return Promise.resolve(createResponse(200, {'Content-Type': ['text/html'] }, '<html><body>' + body + '</body></html>'));
+      break;
+      
+    case 'update':
+      if (name === undefined || url === undefined || name === null || url === null) {
+        status = "error - missing name or url";
+      }
+
+      try {
+        // Check to see if the redirect exists before we attempt to delete it
+        checkID = await edgeKv.getText({ item: name, default_value: undefined });

```

Comment:
This is an anti pattern in EdgeKV. Due to the eventual consistency properties of EdgeKV, there is no guarantee that this will return a value even if the item was created by the same edge server. As such, I suggest collapsing the create and update cases and having the URL checking be tracked by the user outside of EdgeKV or not checked at all. In the latter case you will always create a new short URL or update the existing one if the name exists, but this could happen anyway (and very likley in fact) even if you call getText. So calling getText is both misleading the users and also adding unnecessary latency.